File: baselibrary/src/main/java/com/yzy/baselibrary/widget/imagewatcher/ImageBrowseUtils.java
package com.yzy.baselibrary.widget.imagewatcher;

import android.support.v4.app.FragmentActivity;
import android.view.View;
import android.view.ViewGroup;

/**
 * description :
 *
 * @author : case
 * @date : 2018/8/8 16:17
 */
public class ImageBrowseUtils {
  public static ImageBrowse newInstance(FragmentActivity activity, EnterExitListener listener,
      boolean isShowIndex) {
    return ImageBrowse.Companion.newInstance(activity, listener, isShowIndex);
  }

  public static ImageBrowse newInstance(FragmentActivity activity, boolean isShowIndex) {
    return ImageBrowse.Companion.newInstance(activity, null, isShowIndex);
  }

  /** 是否需要大图处理返回，不需要则返回false */
  public static boolean handleBackPressed(FragmentActivity activity) {
    if (activity != null) {
      ViewGroup view = (ViewGroup) activity.getWindow().getDecorView();
      int count = view.getChildCount();
      for (int i = count - 1; i >= 0; i--) {
        View child = view.getChildAt(i);
        if (child instanceof ImageWatcher) {
          return ((ImageWatcher) child).handleBackPressed();
        }
      }
    }
    return false;
  }
}

File: baselibrary/src/main/java/com/yzy/baselibrary/widget/imagewatcher/ImageBrowseUtils.kt
package com.yzy.baselibrary.widget.imagewatcher

import android.net.Uri
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.FragmentActivity


object ImageBrowseUtils {
    fun newInstance(
        activity: FragmentActivity, listener: EnterExitListener,
        isShowIndex: Boolean, logClickListener: ((v: View, uri: Uri, pos: Int) -> Unit)? = null
    ): ImageBrowse {
        return ImageBrowse.newInstance(activity, listener, isShowIndex, logClickListener)
    }

    fun newInstance(
        activity: FragmentActivity,
        isShowIndex: Boolean,
        logClickListener: ((v: View, uri: Uri, pos: Int) -> Unit)? = null
    ): ImageBrowse {
        return ImageBrowse.newInstance(activity, null, isShowIndex, logClickListener)
    }

    /** 是否需要大图处理返回，不需要则返回false  */
    fun handleBackPressed(activity: FragmentActivity?): Boolean {
        if (activity != null) {
            val view = activity.window.decorView as ViewGroup
            val count = view.childCount
            for (i in count - 1 downTo 0) {
                val child = view.getChildAt(i)
                if (child is ImageWatcher) {
                    return child.handleBackPressed()
                }
            }
        }
        return false
    }
}

